### PR TITLE
complete port of Neuron class

### PR DESCRIPTION
**Pull request by leothorp**

### CODE REVIEW

File: NeuralNetwork/NetworkModels/Neuron.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    public List<Synapse> OutputSynapses { get; set; }
+    public double Bias { get; set; }
+    public double BiasDelta { get; set; }
+    public double Gradient { get; set; }
+    public double Value { get; set; }
+*/
+
+const sum = (arr, transform = x => x) => arr.reduce((acc, curr) => {
+    return acc + transform(curr);
+}, 0);
```
Comment (jnis77diver) [inline NeuralNetwork/NetworkModels/Neuron.js:19]: probably want to move this to a util file. I think i added one in my PR so you can move it there once it's merged.

File: NeuralNetwork/NetworkModels/Neuron.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+
+class Neuron {
+  /* optionally takes an array of Neurons */
+    constructor(inputNeurons = null) {
+        this.Id = uuidv4();
```
Comment (jnis77diver) [inline NeuralNetwork/NetworkModels/Neuron.js:25]: nice, didn't know this npm package existed. thought you had to roll your own Guid generator in JS. b

File: NeuralNetwork/NetworkModels/Neuron.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        this.InputSynapses = [];
+        this.OutputSynapses = [];
+        this.Bias = Network.GetRandom();
+        this.BiasDelta = 0;
+        this.Gradient = 0;
+        this.Value = 0;
```
Comment (jnis77diver) [inline NeuralNetwork/NetworkModels/Neuron.js:31]: in my PR, i made all props camel case (though kept class pascal case). we'll want to decide on which way to go. I think JS convention in classes is pascal case name with camel case props but whatever is cool.

File: NeuralNetwork/NetworkModels/Neuron.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                this.InputSynapses.Add(synapse);   
+            });
+        }
+    }
+
+    //Values & Weights
+    //public virtual double CalculateValue()
+    CalculateValue() {
+        const inputSynapsesValueSum = sum(this.InputSynapses, synapse => synapse.Weight * synapse.InputNeuron.Value);
+        this.Value = Sigmoid.Output(inputSynapsesSum + this.Bias);
```
Comment (jnis77diver) [inline NeuralNetwork/NetworkModels/Neuron.js]: i'm not seeing inputSynapsesSum  anywhere else. should this be inputSynapsesValueSum?

**Review by jnis77diver — COMMENTED** [review]
Few comments

File: NeuralNetwork/NetworkModels/Neuron.js (hunk 1 above)
Comment (leothorp) [reply NeuralNetwork/NetworkModels/Neuron.js:19]: Want to choose one or the other of our `sum` functions to stick with? My motivation for this one was mimicking the c# `Sum`

File: NeuralNetwork/NetworkModels/Neuron.js (hunk 4 above)
Comment (leothorp) [reply NeuralNetwork/NetworkModels/Neuron.js]: yes indeed

**Pull request merged**